lib/chef/provider/user/linux.rb

```diff
@@ -58,6 +58,7 @@ def universal_options
 
         def usermod_options
           opts = []
+          opts << "-u %d" % new_resource.uid if non_unique?
```

Comment:
uh, well, avoid the interpolation entirely:  `opts += [ "-u", new_resource.uid ]`